enarxbot-copy-labels-linked

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/python3
+# SPDX-License-Identifier: Apache-2.0
+
+import bot
+
+import json
+import sys
+import os
+import re
+
+# We don't want to copy all labels on linked issues; ignore those in this subset.
+BLOCKED_LABELS_NAMES = {
+    "meta",
+    "question",
+    "help wanted",
+    "good first issue",
+    "duplicate",
+}
+
+QUERY = '''
+query($id:ID!, $org:String!, $repo:String!, $commitsCursor:String, $prLabelsCursor:String, $issuesCursor:String, $issuesLabelsCursor:String) {
+    node(id:$id) {
+        ... on PullRequest {
+            number
+            id
+            body
+            commits(first:100, after:$commitsCursor) {
+                pageInfo { endCursor hasNextPage }
+                nodes {
+                    commit {
+                        message
+                    }
+                }
+            }
+            labels(first:100, after:$prLabelsCursor) {
+                pageInfo { endCursor hasNextPage }
+                nodes {
+                    name
+                    id
+                }
+            }
+        }
+    }
+    repository(owner:$org, repo:$repo) {
+        issues(first:100, states:OPEN, after:$issuesCursor) {
+            pageInfo { endCursor hasNextPage }
+            nodes {
+                ...on Issue {
+                    number
+                    labels(first:100, after:$issuesLabelsCursor) {
+                        pageInfo { endCursor hasNextPage }
+                        nodes {
+                            name
+                            id
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+'''
+
+QUERY_CURSORS = {
+    "commitsCursor": ["node", "commits"],
+    "prLabelsCursor": ["node", "labels"],
+    "issuesCursor": {
+        "path": ["repository", "issues"],
+        "next": {
+            "issuesLabelsCursor": ["labels"]
+        }
+    }
+}
+
+# Get all issue numbers related to a PR.
+def get_related_issues(body, commits):
+    # Regex to pick out closing keywords.
+    regex = re.compile("(close[sd]?|fix|fixe[sd]?|resolve[sd]?)\s*:?\s+#(\d+)", re.I)
+
+    # Extract all associated issues from closing keyword in PR
+    for verb, num in regex.findall(body):
+        yield int(num)
+
+    # Extract all associated issues from PR commit messages
+    for c in commits:
+        for verb, num in regex.findall(c["commit"]["message"]):
+            yield int(num)
+
+if os.environ["GITHUB_EVENT_NAME"] not in ["pull_request_target"]:
+    sys.exit(0)
```

Comment:
What is this? Where is `pull_request_target` from?

Reply:
`pull_request_target` is an Actions trigger that allows us to listen for PR events (like a new one being submitted) while using the `master` tree of our codebase. This has some nice security benefits, in addition to allowing things like access to shared secrets (which is normally locked down when running on just a regular `pull_request` event).